service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+package fi.nls.oskari.search.channel;
+
+import fi.mml.portti.service.search.ChannelSearchResult;
+import fi.mml.portti.service.search.SearchCriteria;
+import fi.mml.portti.service.search.SearchResultItem;
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.search.channel.SearchChannel;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.PropertyUtil;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Oskari(OgcApiSearchChannel.ID)
+public class OgcApiSearchChannel extends SearchChannel {
```

Comment:
The search is hard-coded for `geographic-names` but the name suggests this is some generic base class for querying OGC API services. Rename the class (and change the ID) to something that is more in line with the functionality. Maybe NLSFIGeographicNamesSearchChannel? I don't feel strongly that OGC API should NOT be part of the name but I don't think it's something that needs to be part of the name as it's more of an implementation detail.